### PR TITLE
Use tokio runtime handle instead of `TaskExecutor` abstraction

### PR DESCRIPTION
**Pull request by bkchr**
Before this pr we had the `TaskExecutor` abstraction which theoretically
allowed that any futures executor could have been used. However, this
was never tested and is currently not really required. Anyone running a
node currently only used tokio and nothing else (because this was hard
coded in CLI). So, this pr removes the `TaskExecutor` abstraction and
relies directly on the tokio runtime handle.

Besides this changes, this pr also makes sure that the http and ws rpc
server use the same tokio runtime. This fixes a panic that occurred when
you drop the rpc servers inside an async function (tokio doesn't like
that a tokio runtime is dropped in the async context of another tokio
runtime).

As we don't use any custom runtime in the http rpc server anymore, this
pr also removes the `rpc-http-threads` cli argument. If external parties
complain that there aren't enough threads for the rpc server, we could
bring support for increasing the thread count of the tokio runtime.

polkadot companion: https://github.com/paritytech/polkadot/pull/3830

### CODE REVIEW

File: client/rpc-servers/src/lib.rs [hunk 1]
```diff
@@ -137,17 +134,18 @@ mod inner {
...
+		tokio_handle: tokio::runtime::Handle,
 	) -> io::Result<http::Server> {
 		let max_request_body_size = maybe_max_payload_mb
 			.map(|mb| mb.saturating_mul(MEGABYTE))
 			.unwrap_or(RPC_MAX_PAYLOAD_DEFAULT);
 
 		http::ServerBuilder::new(io)
-			.threads(thread_pool_size.unwrap_or(HTTP_THREADS))
+			.threads(1)
+			.event_loop_executor(tokio_handle)
```
Comment (bkchr) [inline client/rpc-servers/src/lib.rs:148]: I read the docs and checked the code.

I'm almost sure, but can someone confirm that this is correct?

Comment (niklasad1) [reply client/rpc-servers/src/lib.rs]: /cc @tomusdrw but given https://docs.rs/jsonrpc-http-server/18.0.0/jsonrpc_http_server/struct.ServerBuilder.html#method.threads sounds right to me

Comment (bkchr) [reply client/rpc-servers/src/lib.rs]: By correct I mean that we are still answering requests in parallel.

Comment (niklasad1) [reply client/rpc-servers/src/lib.rs]: yeah, I was referring to

> So it’s also possible to run a multi-threaded server by passing the default tokio::runtime executor to this builder and setting threads to 1.

Comment (bkchr) [reply client/rpc-servers/src/lib.rs]: Yeah, my second comment was just to make my first one more clear xD

Comment (tomusdrw) [reply client/rpc-servers/src/lib.rs]: Yup, I believe so too!

**Review by ordian — APPROVED** [review]
I don't know how to review this properly. Ideally, this PR could use a burnin on a RPC node, but I'm not sure it's warranted.

I understand that we lose the `TaskExecutor` abstraction here, that might worth adding back later, but that's probably out of scope here .

Other than that, looks good to me.

**Comment by bkchr** [comment]
> I don't know how to review this properly. Ideally, this PR could use a burnin on a RPC node, but I'm not sure it's warranted.

This custom http thread number wasn't used by us, I will ping the people that requested it. 

> 
> I understand that we lose the `TaskExecutor` abstraction here, that might worth adding back later, but that's probably out of scope here .

It was currently not used anyway and I think it may also not worked with other executors, because hyper for example requires Tokio. However, as we will refactor a lot of things, this is one of the first things that would have been gone anyway. 

> 
> Other than that, looks good to me.

👍

**Review by niklasad1 — APPROVED** [review]
LGTM, but yeah would good check with users of the `--rpc http threads`

**Comment by bkchr** [comment]
bot merge

**Pull request merged**